dio/lib/src/options.dart

```diff
@@ -80,27 +80,30 @@ class BaseOptions extends _RequestConfig {
           maxRedirects: maxRedirects,
           requestEncoder: requestEncoder,
           responseDecoder: responseDecoder,
-        );
+        ) {
+    this.queryParameters = queryParameters ?? {};
+    this.baseUrl = baseUrl ?? '';
```

Comment:
If you use an initializer list here
```dart
: queryParameters = queryParameters ?? {},
 super();
```

then queryParameters doesn't need to be `late`. The reason why we can't use a default parameter here is that default parameters need to be `const`, which would mean that you can't modify the map afterwards anymore.
(See [the section about constructors in the language tour](https://dart.dev/guides/language/language-tour#constructors) for more information about initializer lists.